### PR TITLE
[stable/traefik] add extraEnv to be passed through to container

### PR DESCRIPTION
**Pull request by rhysm**
Signed-off-by: Rhys Meaclem <rhysmeaclem@gmail.com>

#### What this PR does / why we need it:
Allows users to pass environment variables through to the running traefik container. e.g. http_proxy, https_proxy, no_proxy.

#### Which issue this PR fixes
No known issue

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/traefik/README.md [hunk 1]
```diff
@@ -91,6 +91,7 @@ The following table lists the configurable parameters of the Traefik chart and t
 | `serviceType`                          | A valid Kubernetes service type                                                                                              | `LoadBalancer`                                    |
 | `loadBalancerIP`                       | An available static IP you have reserved on your cloud platform                                                              | None                                              |
 | `startupArguments`                       | A list of startup arguments which are passed to traefik                                                              | `[]`                                              |
+| `extraEnv`                             | Additional environment variables to set on the container                                                                     | `{}`                                              |
```
Comment (dtomcej) [inline stable/traefik/README.md:94]: I like the PR, and it looks good, Im just not sure `extraEnv` is the name I would use for this..

How about:
- additionalEnvVars
- customEnvVars
- envVars

I'm just not sure the `extra` portion adds anything... (and I personally like envVars as the name, but I could be convinced)

Thoughts?

**Review by dtomcej — CHANGES_REQUESTED** [review]

File: stable/traefik/README.md (hunk 1 above)
Comment (rhysm) [reply stable/traefik/README.md:94]: I hadn't put much thought into the naming. It seems the majority of charts in the repo are using `extraEnv` for this purpose so I figured I'd stick with that.

There are 2 charts using `envVars` currently.

In thinking about it now I do prefer all of the suggestions you made over `extraEnv`. If you're not fussed on being different from the other charts then I can change the name to `envVars`.

**Comment by rhysm** [comment]
Closing in favour of #10928 as it's more flexible and will work for my use case too.

**Pull request closed without merging**